CDC_Backend/README.md

```diff
@@ -917,10 +891,236 @@ Response is a Json with these fields
+
+This api is used to submit a students application
+
+Request_Body:
+
+```json
+{
+  "applications_id": "F28IRGGMC3RP8Y8",
+  "student_id": "200010022",
+  "opening_id": "Q54IRZZMC3RP8F6",
+  "additional_info": {
+    "School": "Lions International",
+    "City": "Banglore"
+  },
+  "resume_file_name": "8KIOT3PW1JIS718_CSE-V-SEM.pdf"
+}
+```
+
+> Headers <br>
+> Authorization: "Bearer {tokenID}"
+
+### Response
+
+Response is a Json with these fields
+
+```json
+{
+  "action": "Add Student Application",
+  "message": "Application Added"
+}
+```
+
+### Status Codes
+
+The possible responses for this api request are as follows
+
+| Status Codes    | Possible Messages      |
+| --------------- | ---------------------- |
+| 200 OK          | `Application Added`    |
+| 400 BAD_REQUEST | `Something Went Wrong` |
+
+You may see some different errors which can be seen [here](#common-errors)
+
+---
+
+## `api/admin/generateCSV`
+
+This api is used to generate a CSV file that contains sequential details of all the applications
+
+Request_Body:
+
+```json
+{
+  "opening_id": "Q54IRZZMC3RP8F6"
+}
+```
+
+> Headers <br>
+> Authorization: "Bearer {tokenID}"
+
+### Response
+
+Response is a Json with these fields
+
+```json
+{
+  "action": "Create CSV",
+  "message": "CSV Created",
+  "file": "csv_file_link"
+}
+```
+
+### Status Codes
+
+The possible responses for this api request are as follows
+
+| Status Codes    | Possible Messages      |
+| --------------- | ---------------------- |
+| 200 OK          | `CSV created`          |
+| 400 BAD_REQUEST | `Something Went Wrong` |
+
+You may see some different errors which can be seen [here](#common-errors)
+
+---
+
+## `api/admin/addPPO`
+
+This api is used to add a PPO for a student in the corresponding company
+
+Request_Body:
+
+```json
+{
+  "company_name": "Geeks for Geeks",
+  "compensation_gross": "1000000",
+  "offer_accepted": "true",
+  "student_id": "200010022",
+  "designation": "Technical Content Writer",
+  "tier": "4"
+}
+```
+
+### Response
+
+Response is a Json with these fields
+
+```json
+{
+  "action": "Add PPO",
+  "message": "PPO Added"
+}
+```
+
+### Status Codes
+
+The possible responses for this api request are as follows
+
+| Status Codes    | Possible Messages      |
+| --------------- | ---------------------- |
+| 200 OK          | `PPO Added`            |
+| 400 BAD_REQUEST | `Something Went Wrong` |
+
+You may see some different errors which can be seen [here](#common-errors)
+
+---
+
+## `api/admin/getStudentApplication`
+
+This api is used to fetch the application details of the required student
+
+Request_Body:
+
+```json
+{
+  "student_id": "200010022",
+  "opening_id": "Q54IRZZMC3RP8F6"
+}
+```
+
+> Headers <br>
+> Authorization: "Bearer {tokenID}"
+
+### Response
+
+Response is a Json with these fields
+
+```json
+{
+  "action": "Get Student Application",
+  "application_found": "true",
+  "application_info": {
+    "id": "F28IRGGMC3RP8Y8",
+    "additional_info": ,
```

Comment:
additional info is a json objects with keys as entered by company for example company may 12th marks i.e `"additional_info":{"12marks":98}`.

Reply:
field added